lib/packwerk/formatters/offenses_formatter.rb

```diff
@@ -6,11 +6,13 @@ module Formatters
     class OffensesFormatter
       include Packwerk::OffensesFormatter
 
+      IDENTIFIER = T.let("default", String)
+
       extend T::Sig
 
-      sig { params(style: OutputStyle).void }
-      def initialize(style: OutputStyles::Plain.new)
-        @style = style
+      sig { returns(OutputStyle) }
+      def style
```

Comment:
The `OffensesFormatter` interface calls `new` with these.

Using a YML file as the vehicle for configuration makes it hard to allow the client to specify their own `initialize` method.

Note that if we had a configuration option that looked like:
```ruby
Packwerk.configure do |config|
  config.formatters = [ MyFormatter.new(i_can_pass_in_whatever_here) ]
end
```

This would not be as much of a problem. However, since our implementation for loading in extensions is YML, we need to make the classes that implement the interfaces very simple. Therefore, I split this up into two classes: one that is the "default" that is colored, and one that right now is only used in tests which is the "plain" formatter. This way the class can be initialized and registered by the interface without an issue.

Reply:
Using an abstract class instead where we control the initializer might help this situation. However, since people control the implementation of their formatter, we can assume they will not need to pass in anything special as they can memoize or inline it as needed.

Reply:
If we re-define the style in the formatter, does it make sense to pass the style into the CLI object anymore? Maybe we should remove that.

I'm also not 100% sure this should be public. It seems optional.

Reply:
Yeah, we still need style as a CLI arg for now for the progress formatter. However, you're right we don't need `style` as a public method of this formatter – made it private!